src/imitation/algorithms/bc.py

```diff
@@ -123,28 +130,20 @@ def train(
             the end of each epoch. Will receive all locals from this function as
             dictionary argument (!!).
         """
-        epoch_iter = trange(n_epochs, desc="BC epoch")
-        for epoch_num in epoch_iter:
-            total_batches = self.expert_dataset.n_samples // self.batch_size
-            batch_iter = self.expert_dataset.iterate_once(self.batch_size)
-            tq_iter = tqdm(
-                batch_iter, total=total_batches, desc="pol step", leave=False
-            )
-            loss_ewma = None
-            for batch_dict in tq_iter:
+        assert self.batch_size >= 1
+        samples_so_far = 0
+        for epoch_num in trange(n_epochs, desc="BC epoch"):
+            while samples_so_far < (epoch_num + 1) * self.expert_dataset.size():
+                batch_dict = self.expert_dataset.sample(self.batch_size)
+                assert len(batch_dict["obs"]) == self.batch_size
+                samples_so_far += self.batch_size
```

Comment:
I removed some loss stdout display here. But I have another branch where I replaced it with `logging.log_kv` and other good stuff to make Tensorboard plots and stdout tables. So that's why I didn't bother reattaching the old loss_ewma stuff after I changed BC things around in this PR.